core/src/banking_stage/committer.rs

```diff
@@ -23,6 +23,83 @@ use {
     std::{collections::HashMap, sync::Arc},
 };
 
+pub(crate) static FIREDANCER_COMMITTER: std::sync::atomic::AtomicU64 = std::sync::atomic::AtomicU64::new(0);
+
+#[no_mangle]
+pub extern "C" fn fd_ext_bank_pre_balance_info( bank: *const std::ffi::c_void, txns: *const std::ffi::c_void, txn_count: u64 ) -> *mut std::ffi::c_void {
+    use solana_sdk::transaction::SanitizedTransaction;
+    use std::borrow::Cow;
+    use std::sync::atomic::Ordering;
+
+    let txns = unsafe {
+        std::slice::from_raw_parts(txns as *const SanitizedTransaction, txn_count as usize)
+    };
+    let bank = bank as *const Bank;
+    unsafe { Arc::increment_strong_count(bank) };
+    let bank = unsafe { Arc::from_raw( bank as *const Bank ) };
+
+    let lock_results = txns.iter().map(|_| Ok(()) ).collect::<Vec<_>>();
+    let batch = TransactionBatch::new(lock_results, bank.as_ref(), Cow::Borrowed(txns));
+
+    while FIREDANCER_COMMITTER.load(Ordering::Acquire) == 0 {
+        std::hint::spin_loop();
+    }
```

Comment:
Is there actually a race condition this protects against? I'm okay with leaving it, but FD would have to get a bank before `FIREDANCER_COMMITTER` gets initialized. Can that happen?